### PR TITLE
feat!: Separate logical MySQL host from connected host

### PR DESCRIPTION
**Pull request by arielvalentin**
MySQL server instances have a logical host name and an instance node names.

This is particularly useful when running in clustered mode where a logical hostname is used to connect to a cluster but you also want to record what specific node is executing your query.

Trilogy exposes the instance node name via `@connected_host` as well as the configured `host` name in its `connection_options`.

This breaking change ensures that the `net.peer.name` is always set to the logical host name, while the individual instance name is being recorded in a separate attribute.

cc: @cschleiden

### CODE REVIEW

File: instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb [hunk 1]
```diff
@@ -68,11 +63,12 @@ def query(sql)
           def client_attributes(sql)
             attributes = {
               ::OpenTelemetry::SemanticConventions::Trace::DB_SYSTEM => 'mysql',
-              ::OpenTelemetry::SemanticConventions::Trace::NET_PEER_NAME => net_peer_name
+              ::OpenTelemetry::SemanticConventions::Trace::NET_PEER_NAME => connection_options.fetch(:host, 'unknown sock'),
             }
 
             attributes[::OpenTelemetry::SemanticConventions::Trace::DB_NAME] = database_name if database_name
             attributes[::OpenTelemetry::SemanticConventions::Trace::PEER_SERVICE] = config[:peer_service] unless config[:peer_service].nil?
+            attributes['db.mysql.instance.host.name'] = @connected_host if defined?(@connected_host)
```
Comment (arielvalentin) [inline instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb:71]: I have no idea what this be named and I am struggling here. 

What do you think @adrianna-chang-shopify @cschleiden ?

Comment (arielvalentin) [reply instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb:71]: cc: @open-telemetry/specs-semconv-approvers I could use some of your input/feedback here as well.

Comment (cschleiden) [reply instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb:71]: I see https://github.com/open-telemetry/opentelemetry-ruby/blob/main/semantic_conventions/lib/opentelemetry/semantic_conventions/trace.rb#L71 

looking at https://github.com/open-telemetry/opentelemetry-specification/pull/3402, maybe
`db.mysql.instance.address` or `db.mysql.address`? But `db.mysql.instance.host.name` would also work for me

**Review by cschleiden — APPROVED** [review]
Thanks for this change!

File: instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb (hunk 1 above)
Comment (arielvalentin) [reply instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb:71]: Nice find!

File: instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb [hunk 2]
```diff
@@ -68,11 +63,12 @@ def query(sql)
           def client_attributes(sql)
             attributes = {
               ::OpenTelemetry::SemanticConventions::Trace::DB_SYSTEM => 'mysql',
-              ::OpenTelemetry::SemanticConventions::Trace::NET_PEER_NAME => net_peer_name
+              ::OpenTelemetry::SemanticConventions::Trace::NET_PEER_NAME => connection_options.fetch(:host, 'unknown sock'),
```
Comment (cschleiden) [inline instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb:66]:
```suggestion
              ::OpenTelemetry::SemanticConventions::Trace::NET_PEER_NAME => connection_options.fetch(:host, 'unknown sock')
```

to make the linter happy

File: instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb (hunk 1 above)
Comment (arielvalentin) [reply instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb:71]: cc: @andyedison

**Review by ericmustin — APPROVED** [review]
lgtm, may wanna confirm with known users of the instrumentation (cc @adrianna-chang-shopify ) that this works for them.

**Comment by ericmustin** [comment]
ah @arielvalentin looks like a couple typos, my apols on the early approval, that aside this lgtm

**Review by cschleiden — APPROVED** [review]

**Review by plantfansam — APPROVED** [review]
Love this! As a trilogy instrumentation user, we can handle the churn in attr name 😄.

**Pull request merged**